src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.Unix.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Security;
+using System.Security.Principal;
+using System.Threading;
+using System.ComponentModel;
+using System.Security.Authentication;
+using System.Security.Authentication.ExtendedProtection;
+
+namespace System.Net.Security
+{
+    //
+    // The class maintains the state of the authentication process and the security context.
+    // It encapsulates security context and does the real work in authentication and
+    // user data encryption with NEGO SSPI package.
+    //
+    // This is part of the NegotiateStream PAL.
+    //
+    internal class NegoState
+    {
+        internal const int c_MaxReadFrameSize = 64 * 1024;
+        internal const int c_MaxWriteDataSize = 63 * 1024; // 1k for the framing and trailer that is always less as per SSPI.
```

Comment:
The way the code was split these are needed by InternalNegotiateStream. I'd rather not make too many changes since I don't know the form this will have for other platforms.

/cc: @vijaykota @ericeil